lib/model/internal_link.dart

```diff
@@ -205,6 +206,9 @@ enum _NarrowOperator {
   // 'dm' is new in server-7.0; means the same as 'pm-with'
   dm,
   near,
+  // cannot use lowerCamelCase `with` as it is a reserved keyword in Dart
+  // ignore: constant_identifier_names
+  With,
```

Comment:
Fun.

Let's call it `with_`. The leading capital reads to me like it's a different kind of thing — like not an enum value at all, but a class or something like a class.

Reply:
It will generate the same "with_" string value in the enum map which is not compatible with the API!
<img width="329" alt="with enum" src="https://github.com/zulip/zulip-flutter/assets/59946442/bfda8d47-3b0c-44ac-bec6-f1ec3e18fa39">

Reply:
Ah, that can be adjusted, though. The JSON serialization is all nicely customizable with more annotations — for a value of an enum, it's `@JsonValue`:
https://pub.dev/documentation/json_annotation/4.9.0/json_annotation/JsonValue-class.html

Reply:
Done! Thank you.

Also, as this is my first time using the `json_annotation` package, I have a hard time working with it:
When I run the `dart run build_runner watch --delete-conflicting-outputs` command, all of the other auto-generated files are also modified, such as `lib/api/model/events.g.dart`, `lib/api/route/account.g.dart`, and more — and some are deleted such as `lib/host/android_notifications.g.dart`. So far, I have used `git restore file` to reject changes to those files, but I am not sure if I am doing things correctly from the start. Also, the files that these modified files are used in, gives an error like this:
```
Target of URI hasn't been generated: './android_notifications.g.dart'.
Try running the generator that will generate the file referenced by the URI.dart[uri_has_not_been_generated]
```
Even though, I run the previous command.

Reply:
(For cross-reference: this [turned out](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.20json_annotation.20unexpected.20behavior/near/1824402) to be a newlines issue, of CRLF vs. LF.)